app/scripts/directives/datepicker.js

```diff
@@ -10,25 +10,51 @@ angular
       scope: {
         localModel: '=model',
         ngDisabled: '=',
+        graduationDate: '=graduationDate',
       },
       controller: function ($timeout, $scope) {
         $scope.updateTimeStamp = function (timestamp) {
           $scope.$apply(function () {
-            $scope.localModel = moment(timestamp).format('YYYY-MM-DD HH:mm:ss');
+            $scope.localModel = moment(new Date(timestamp)).format(
+              'YYYY-MM-DD HH:mm:ss',
+            );
           });
         };
       },
       link: function (scope, element) {
         var datePickerElement = angular.element(element).find('.datepicker');
-        var initialDate = scope.localModel
-          ? moment(scope.localModel).format('MM/DD/YYYY hh:mm A')
-          : null;
+        var initialDate =
+          scope.localModel && scope.graduationDate
+            ? scope.localModel
+            : scope.localModel
+            ? moment(new Date(scope.localModel)).format('MM/DD/YYYY hh:mm A')
+            : null;
+        let dateOptions = scope.graduationDate
+          ? {
+              viewMode: 'years',
+              format: 'MMMM YYYY',
+              defaultDate: initialDate,
+              useCurrent: false,
+              extraFormats: [
+                'MM/YY',
+                'MM/YYYY',
+                'MM-YY',
+                'MM-YYYY',
+                'MMM-YYYY',
+                'MMMM-YYYY',
+              ],
+            }
+          : {
+              defaultDate: initialDate,
+            };
```

Comment:
You could rename graduationDate to `dateFormat` and use it in the code as the format? Just a thought.
```suggestion
          scope.localModel ? moment(new Date(scope.localModel)).format(`${scope.dateFormat} hh:mm A`)
            : null;
        let dateOptions = scope.dateFormat === 'MMMM-YYYY'
          ? {
              viewMode: 'years',
              format: dateFormat,
              defaultDate: initialDate,
              useCurrent: false,
              extraFormats: [
                'MM/YY',
                'MM/YYYY',
                'MM-YY',
                'MM-YYYY',
                'MMM-YYYY',
                'MMMM-YYYY',
              ],
            }
          : {
              defaultDate: initialDate,
            };
```

Reply:
I think that could be a little confusing because people might think that setting dateFormat could change the format to whatever they want. When there is actually only 1 value they can provide.